test/test_distributions.py

```diff
@@ -2500,6 +2545,8 @@ def __init__(self, probs):
         # e.g. bernoulli[1] varies row-wise; that way we test all param pairs.
         bernoulli = pairwise(Bernoulli, [0.1, 0.2, 0.6, 0.9])
         binomial30 = pairwise(Binomial30, [0.1, 0.2, 0.6, 0.9])
+        binomial_vectorized_count = (Binomial(torch.tensor([3., 4.]), torch.tensor([0.4, 0.6])),
```

Comment:
This is consistent with `torch.bernoulli()` returning a float tensor. When I tried making all the morally-integer datatypes in torch.distributions into `LongTensor`s, there were lots of incompatibilities and inconsistencies with the rest of PyTorch. The only remaining distribution for which we use `LongTensor`s is `Categorical`. #WeirdAllTheWayDown

Reply:
hmm..the issue is that we have operations between `counts` and `probs`. This wasn't an issue when counts was a python numeric type, but if we have it as a `LongTensor`, we'll just need to type cast it as float when we do these operations, or ideally just once in the constructor. I suppose that will be more idiomatic?

Reply:
> When I tried making all the morally-integer datatypes in torch.distributions into LongTensors 

I think this will be so much easier when (or if) we have implicit type conversion for various operations. We can then support a much saner looking API.

Reply:
I think `torch.distributions.Poisson` should be consistent with `torch.poisson`. Currently `torch.poisson` returns floats:
```py
>>> torch.poisson(torch.tensor(0.5))
tensor(0.)
```

Reply:
Yes that's not great, but it's not the same I think. It returns them as random samples, while what we have here is a distribution parameter. Also, this is legacy stuff, and we will probably want to change that once we add cross-type operations.